x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -6,21 +6,34 @@
  */
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import createContainer from 'constate';
-import { useCallback } from 'react';
+import { useCallback, useEffect } from 'react';
 import { buildEsQuery, Filter, Query, TimeRange } from '@kbn/es-query';
 import { debounce } from 'lodash';
 import type { SavedQuery } from '@kbn/data-plugin/public';
 import type { InfraClientStartDeps } from '../../../../types';
 import { useMetricsDataViewContext } from './use_data_view';
 import { useSyncKibanaTimeFilterTime } from '../../../../hooks/use_kibana_timefilter_time';
-import { useHostsUrlState, INITIAL_DATE_RANGE } from './use_hosts_url_state';
+import { useHostsUrlState, INITIAL_DATE_RANGE, HostsState } from './use_hosts_url_state';
+
+const buildQuerySubmittedPayload = (hostState: HostsState) => {
+  const { panelFilters, filters, dateRange, query: queryObj } = hostState;
+
+  return {
+    control_filters: panelFilters.map((filter) => JSON.stringify(filter)),
+    filters: filters.map((filter) => JSON.stringify(filter)),
```

Comment:
`Filter` type has `state$` and `meta` properties that we might not be interested in, for telemetry purposes. 

How about sending only the query? 

```
control_filters: JSON.stringify(panelFilters.map((filter) => (filter.query)),
filters: JSON.stringify(filters.map((filter) => filter.query))),
```

This way, we'll get something like:

```
[{
  match_phrase: { host.name: "metricbeat" }
}]
````

Reply:
This was my first attempt, I had the same opinion as you! However, I noticed in some cases there are details relevant to the Filter which are not in the query.
For example, I noticed that a negated control panel is not marked in the query, but adds a `meta` field `negated: true`, that's why I saw the need to include the whole filter to avoid misleading filters.

Reply:
I see. Indeed.I don't know how useful that information actually is, but let's try sending the whole object and see how it goes.